framework/src/media/MediaRecorderImpl.cpp

```diff
@@ -171,33 +178,58 @@ recorder_result_t MediaRecorderImpl::stop()
 
 recorder_result_t MediaRecorderImpl::pause()
 {
-	lock_guard<mutex> lock(mCmdMtx);
+	std::lock_guard<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaRecorderImpl::pause()\n");
 	RecorderWorker& mrw = RecorderWorker::getWorker();
 	mrw.getQueue().enQueue(&RecorderWorker::pauseRecorder, &mrw, shared_from_this());
 
 	return RECORDER_OK;
 }
 
-recorder_result_t MediaRecorderImpl::getVolume()
+int MediaRecorderImpl::getVolume()
 {
-	lock_guard<mutex> lock(mCmdMtx);
+	std::lock_guard<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaRecorderImpl::getVolume()\n");
-	return RECORDER_OK;
+
+	if (mCurState == RECORDER_STATE_IDLE || mCurState == RECORDER_STATE_NONE) {
+		meddbg("MediaRecorderImpl::getVolume() - mCurState == RECORDER_STATE_IDLE || mCurState == RECORDER_STATE_NONE\n");
+		return 0;
+	}
+
+	return mCurVolume;
 }
 
-void MediaRecorderImpl::setVolume(int vol)
+recorder_result_t MediaRecorderImpl::setVolume(int vol)
 {
-	lock_guard<mutex> lock(mCmdMtx);
+	std::lock_guard<std::mutex> lock(mCmdMtx);
 	medvdbg("MediaRecorderImpl::setVolume(int vol)\n");
+
+	if (mCurState == RECORDER_STATE_IDLE || mCurState == RECORDER_STATE_NONE) {
+		meddbg("MediaRecorderImpl::setVolume(int vol) - mCurState == RECORDER_STATE_IDLE || mCurState == RECORDER_STATE_NONE\n");
+		return RECORDER_ERROR;
+	}
+
```

Comment:
Please consider below, to reduce call api in audiomanager
if (mCurvolume == vol) {
      medvdbg("volume is same, hence return OK\n");
      return RECORDER_OK;
}

Reply:
audio_manager not yet don't support mic volume.